client-lite/src/util/do_persistence.cpp

```diff
@@ -27,7 +27,11 @@ const std::string& GetConfigDirectory()
 
 const std::string& GetSDKConfigFilePath()
 {
+#ifdef DO_BUILD_FOR_SNAP
+    static std::string configFilePath(DO_CONFIG_DIRECTORY_PATH "/configs/sdk-config.json");
```

Comment:
I would rather not have different paths like this. Instead, I think it is ok to allow write access to the entire /etc/deliveryoptimization-agent folder.